### ISSUE TITLE
[Bug] Ctrl + V does not paste image from clipboard

### ISSUE THREAD

**Issue by Harmon758** [AppFlowy-IO/AppFlowy#3263]
### Bug Description

Attempting to paste an image from clipboard using Ctrl + V doesn't seem to work.

### How to Reproduce

- Copy an image into clipboard
- Attempt to paste into an AppFlowy document using Ctrl + V
- Observe image not being pasted
- Right click in the document -> Paste
- Observe image pasted

### Expected Behavior

Ctrl + V should also work to paste an image from clipboard.

### Operating System

Microsoft Windows [Version 10.0.19045.3086]

### AppFlowy Version(s)

0.3.0

**Comment by LucasXu0** [comment]
@Harmon758 Have you tried using 'Ctrl+V' to paste some plain text?

**Comment by Harmon758** [comment]
Yes, pasting text works fine.

**Comment by LucasXu0** [comment]
I can't reproduce it on my Windows. Can you help paste your image on this website and post the format information here?

https://evercoder.github.io/clipboard-inspector/

**Comment by Harmon758** [comment]
![image](https://github.com/AppFlowy-IO/AppFlowy/assets/9403740/603f4adb-58dd-49b5-8201-341d354935fd)

**Comment by Harmon758** [comment]
Another example, taking a screenshot with ShareX:
![image](https://github.com/AppFlowy-IO/AppFlowy/assets/9403740/abdc1118-b3db-4f9e-9fa7-0282cacc1100)

**Comment by Harmon758** [comment]
I've been able to get it working by copying the image file itself.
Ironically, copying from Firefox's screenshot functionality seems to work as well.

**Comment by LucasXu0** [comment]
Yes, @Harmon758, I can reproduce now. If the image is sourced from the network, the HTML will also be populated in the pasteboard.

**Comment by Harmon758** [comment]
I don't think taking a screenshot with ShareX, as in https://github.com/AppFlowy-IO/AppFlowy/issues/3263#issuecomment-1691303873, has any HTML included, so that might not be the only condition where it doesn't work?

**Comment by LucasXu0** [comment]
> I don't think taking a screenshot with ShareX, as in [#3263 (comment)](https://github.com/AppFlowy-IO/AppFlowy/issues/3263#issuecomment-1691303873), has any HTML included, so that might not be the only condition where it doesn't work?

I have fixed it. It's related to the paste order.

**Comment by leni8ec** [comment]
Why closed?

Pasting image by `Win + V` and `Ctrl + V` - not worked in 0.7.9 (Windows and MacOS)

**Comment by afifurrohman-id** [comment]
> Pasting image by `Win + V` and `Ctrl + V` - not worked in 0.7.9 (Windows and MacOS)

Same here 0.8.3 still not work (Windows 11, Android 14)

**Comment by gus-simon** [comment]
I was able to make the Ctrl + V work from when you get a screenshot using ShareX as an example, by setting to `true` the `UseAlternativeClipboardCopyImage` option under `Application Settings`, `Advanced` inside ShareX configuration.

![Image](https://github.com/user-attachments/assets/a9bea5ab-21dc-47f0-a2c6-8555de21df5e)

This points to me that the AppFlowy issue is related to how it handles pasting "PNG" images without a 32 bit "DIB" (I didn't check AppFlowy code to see which library is used to deal with pasting images from clipboards, etc., but an alternative there could be a good way to go :) ).
